ios/Classes/HyperTrackPlugin.m

```diff
@@ -120,78 +128,54 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
         result(nil);
         return;
     }
-
     
-    result(FlutterMethodNotImplemented);
-}
-
-- (void)onSdkError:(NSNotification*)notification {
-    NSLog(@"%s notification %@", __PRETTY_FUNCTION__, notification);
-    NSError* error = [notification hyperTrackTrackingError];
-    if (error == nil) return;
-
-    NSLog(@"Got error %@", error);
-    _eventSink([self convertErrorToMessage:error]);
+    if ([@"isTracking" isEqualToString:call.method]) {
+        result([NSNumber numberWithBool:[self.hyperTrack isTracking]]);
+        return;
+    }
     
+    if([@"getLatestLocation" isEqualToString:call.method]) {
+        result([self.hyperTrack location]);
+        return;
+    }
+    
+    if([@"setAvailability" isEqualToString:call.method]) {
+        NSNumber *available = call.arguments;
+        [self.hyperTrack setAvailability:(available.boolValue ? HTAvailabilityAvailable : HTAvailabilityUnavailable)];
+        result(nil);
+        return;
+    }
+    
+    if([@"getAvailability" isEqualToString:call.method]) {
+        if([self.hyperTrack availability] == HTAvailabilityAvailable) {
+            result(@"available");
+        } else {
+            result(@"unavailable");
+        }
+        
+        return;
+    }
+    
+    result(FlutterMethodNotImplemented);
 }
 
-- (NSString *)convertErrorToMessage:(NSError *)error {
++ (NSString *)convertErrorToMessage:(NSError *)error {
     switch ([error code]) {
-            case HTRestorableErrorLocationPermissionsDenied:
-            case HTRestorableErrorLocationServicesDisabled:
-            case HTRestorableErrorMotionActivityServicesDisabled:
-            case HTUnrestorableErrorMotionActivityPermissionsDenied:
-            case HTFatalErrorProductionMotionActivityPermissionsDenied:
-                return @"permissions_denied";
-            case HTRestorableErrorTrialEnded:
-            case HTRestorableErrorPaymentDefault:
-               return @"auth_error";
-            case HTUnrestorableErrorInvalidPublishableKey:
-            case HTFatalErrorDevelopmentPublishableKeyIsEmpty:
-                return @"publishable_key_error";
+        case HTRestorableErrorLocationPermissionsDenied:
+        case HTRestorableErrorLocationServicesDisabled:
+        case HTRestorableErrorMotionActivityServicesDisabled:
+        case HTUnrestorableErrorMotionActivityPermissionsDenied:
+        case HTFatalErrorProductionMotionActivityPermissionsDenied:
+            return @"permissions_denied";
```

Comment:
it'd be good to know which permission was denied, imo none of these errors should fall through, but have a dedicated message